### PR TITLE
Added 3 APIs for Flatten function

### LINKED ISSUE #292: `flatten` lacks overloads
`flatten` only has a `ColumnSelector` DSL and lacks the other 3 abilities to select columns (column names, column accessors, KProperties)

### PR DESCRIPTION
**Pull request by zaleslaw**
Fixed #292

### CODE REVIEW

**Comment by zaleslaw** [comment]
@koperagen @Jolanrensen I have a question: why do we miss KProperties as an Access API in the documentation (I could not find tabs with that kind of access). Is it related that it requires a plugin and it is somehow "difficult" for a user?

I also suggest updating the documentation in PRs together with the API changes as a part of triad (code, tests, docs)

File: core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/flatten.kt [hunk 1]
```diff
@@ -1,6 +1,9 @@
 package org.jetbrains.kotlinx.dataframe.api
 
 import io.kotest.matchers.shouldBe
+import org.jetbrains.kotlinx.dataframe.DataRow
+import org.jetbrains.kotlinx.dataframe.annotations.DataSchema
+import org.jetbrains.kotlinx.dataframe.samples.api.TestBase
```
Comment (Jolanrensen) [inline core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/flatten.kt:6]: Unused import

File: core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/flatten.kt [hunk 2]
```diff
@@ -13,6 +16,41 @@ class FlattenTests {
...
+    fun `flatten access APIs`() {
+        val df = dataFrameOf("a", "b", "c")(1, 2, 3)
+        val grouped = df.group("a", "b").into("d")
+
+        // String API
+        grouped.flatten("d") shouldBe df
+        val castedGroupedDF = grouped.cast<Grouped>()
+
+        // KProperties API
+        castedGroupedDF.flatten (Grouped::d) shouldBe df
```
Comment (Jolanrensen) [inline core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/flatten.kt:41]: random space

**Comment by zaleslaw** [comment]
Not, probably I wrote unclear. We have 4 apis but on many pages in operations we have 3 tabs only without Kproperties

File: docs/StardustDocs/topics/flatten.md [hunk 3]
```diff
@@ -11,13 +11,36 @@ flatten  [ { columns } ]
 Columns after flattening will keep their original names. Potential column name clashes are resolved by adding minimal possible name prefix from ancestor columns.
 
 <!---FUN flatten-->
```
Comment (Jolanrensen) [inline docs/StardustDocs/topics/flatten.md:13]: you renamed this function to `flatten_properties` in Modify.kt

We use Korro to check that samples in stardust are actually compilable. So make sure that all examples in markdown are surrounded like
`<!---FUN someFunctionName-->` 
your example, which is between // SampleStart and // SampleEnd comments in samples/api/Modify.kt inside a function with the same name.
`<!---END-->`

**Review by Jolanrensen — CHANGES_REQUESTED** [review]

**Comment by koperagen** [comment]
> Not, probably I wrote unclear. We have 4 apis but on many pages in operations we have 3 tabs only without Kproperties

I think it was done deliberately, but might be discussed. There's not much value in using this API outside its rather narrow use case, like described here https://kotlin.github.io/dataframe/kpropertiesapi.html. So using only 3 APIs in samples was probably an attempt to find a balance between providing this feature for advanced use cases and not confusing people as to what to use.

File: docs/StardustDocs/topics/flatten.md (hunk 3 above)
Comment (zaleslaw) [reply docs/StardustDocs/topics/flatten.md:13]: Template looks like: name of function underscore acess api

**Comment by zaleslaw** [comment]
@Jolanrensen fixed the review

File: docs/StardustDocs/topics/flatten.md (hunk 3 above)
Comment (Jolanrensen) [reply docs/StardustDocs/topics/flatten.md:13]: Oh now I see! Didn't know this was possible. Cool :)

**Review by Jolanrensen — APPROVED** [review]

**Pull request merged**